src/App.tsx

```diff
@@ -15,48 +16,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 'none',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case 'none':
+        return 0;
+      case 'alphabet':
+        return a.localeCompare(b);
+      case 'length':
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const currentReverse = () => {
+    setIsReversed(value => !value);
+  };
+
+  const data = getReorderedGoods(
+    goodsFromServer, { sortType, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            classNames(
+              'button is-info',
+              { 'is-light': sortType !== 'alphabet' },
+            )
+          }
+          onClick={() => {
+            setSortType(SortType.ALPHABET);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={
+            classNames(
+              'button is-success',
+              { 'is-light': sortType !== 'length' },
+            )
+          }
+          onClick={() => {
+            setSortType(SortType.LENGTH);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={
+            classNames(
+              'button is-warning',
+              { 'is-light': !isReversed },
+            )
+          }
+          onClick={() => {
+            currentReverse();
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {
+          (sortType !== 'none' || isReversed)
+            && (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={() => {
+                  setSortType(SortType.NONE);
+                  setIsReversed(false);
+                }}
+              >
+                Reset
+              </button>
+            )
+        }
```

Comment:
the piece can be changed for better readability
```suggestion
        {(sortType !== 'none' || isReversed) && (
          <button
            type="button"
            className="button is-danger is-light"
            onClick={() => {
              setSortType(SortType.NONE);
              setIsReversed(false);
            }}
          >
            Reset
          </button>
        )}
```